### ISSUE TITLE
Sculk Shrieker Agitation system

### ISSUE THREAD

**Issue by MerpisMe** [Osmiooo/wildmod#71]
This may be a huge stretch, but if possible I feel like Sculk Shriekers should use a similar suspicion system to the Warden to decide when to spawn it. Essentially, once the suspicion level (call it something like "shrieker agitation") reaches a certain value, it spawns the warden. Of course, different kinds of triggers would be worth higher values (where stepping directly on a shrieker is the highest and throwing a projectile at it is the lowest).

This should hopefully work as a decently functioning temporary system instead of the current constant number of shrieks needed to summon the warden, and this would also solve the issue with bats being able to summon it. 

Finally, if possible, maybe make sculk sensors send the vibration _type_ to the shrieker when activating it. The only difference would be that sculk sensors would multiply the value of the vibration type by something like 2. This should make it so that things like bats can't cause the warden to spawn. In fact, maybe just outright disallow bats to activate the shrieker via sculk sensors if sending the entity type would be too much work.

TLDR;
- [x] shriekers use an agitation system to decide when the warden should spawn
- [x] sensors send the vibration type to the shrieker when activating it and multiplies the value of the type by 2
- [x] #77
- [x] increase the agitation value of every type of vibration based on the difficulty (finally difficulty actually dictates how difficult the game is!!)

**Comment by AViewFromTheTop** [comment]
Do you know how it ends? Do you know when to stop?
Come out see the light, A View From The Top...
Come out, spinning 'round, cause I can't allow...
Come out and see the view, A View From The Top.

**Comment by AViewFromTheTop** [comment]
> Do you know how it ends? Do you know when to stop? Come out see the light, A View From The Top... Come out, spinning 'round, cause I can't allow... Come out and see the view, A View From The Top.

In other words it's a complex idea but I like it

**Comment by MerpisMe** [comment]
> > Do you know how it ends? Do you know when to stop? Come out, see the light... A View From The Top. Come out, spinning 'round, cause I can't allow... Come out and see the view, A View From The Top.
> 
> In other words it's a complex idea but I like it

i was hesitant to post it because i was worried it might be EXTREMELY difficult, so i'm relieved to see you're on par with the concept of it

**Comment by AViewFromTheTop** [comment]
> > > Do you know how it ends? Do you know when to stop? Come out, see the light... A View From The Top. Come out, spinning 'round, cause I can't allow... Come out and see the view, A View From The Top.
> > 
> > 
> > In other words it's a complex idea but I like it
> 
> i was hesitant to post it because i was worried it might be EXTREMELY difficult, so i'm relieved to see you're on par with the concept of it oh, also, i just figured out i can add checkmark boxes, so i've gone ahead and added them to the suggestions with multiple things on them-- that way, you can easily keep track of what you've done

btw, what should the max value be?

**Comment by MerpisMe** [comment]
> > > > Do you know how it ends? Do you know when to stop? Come out, see the light... A View From The Top. Come out, spinning 'round, cause I can't allow... Come out and see the view, A View From The Top.
> > > 
> > > 
> > > In other words it's a complex idea but I like it
> > 
> > 
> > i was hesitant to post it because i was worried it might be EXTREMELY difficult, so i'm relieved to see you're on par with the concept of it oh, also, i just figured out i can add checkmark boxes, so i've gone ahead and added them to the suggestions with multiple things on them-- that way, you can easily keep track of what you've done
> 
> btw, what should the max value be?

20 sound good?

**Comment by AViewFromTheTop** [comment]
> > > > > Do you know how it ends? Do you know when to stop? Come out, see the light... A View From The Top. Come out, spinning 'round, cause I can't allow... Come out and see the view, A View From The Top.
> > > > 
> > > > 
> > > > In other words it's a complex idea but I like it
> > > 
> > > 
> > > i was hesitant to post it because i was worried it might be EXTREMELY difficult, so i'm relieved to see you're on par with the concept of it oh, also, i just figured out i can add checkmark boxes, so i've gone ahead and added them to the suggestions with multiple things on them-- that way, you can easily keep track of what you've done
> > 
> > 
> > btw, what should the max value be?
> 
> 20 sound good?

yeah. Also I just want to add that this will still end up applying to ALL shriekers in the world as opposed to just one- otherwise it'd be a chaotic mess if I tried to get it to work with Warden summoning properly.

**Comment by MerpisMe** [comment]
> > > > > > Do you know how it ends? Do you know when to stop? Come out, see the light... A View From The Top. Come out, spinning 'round, cause I can't allow... Come out and see the view, A View From The Top.
> > > > > 
> > > > > 
> > > > > In other words it's a complex idea but I like it
> > > > 
> > > > 
> > > > i was hesitant to post it because i was worried it might be EXTREMELY difficult, so i'm relieved to see you're on par with the concept of it oh, also, i just figured out i can add checkmark boxes, so i've gone ahead and added them to the suggestions with multiple things on them-- that way, you can easily keep track of what you've done
> > > 
> > > 
> > > btw, what should the max value be?
> > 
> > 
> > 20 sound good?
> 
> yeah. Also I just want to add that this will still end up applying to ALL shriekers in the world as opposed to just one- otherwise it'd be a chaotic mess if I tried to get it to work with Warden summoning properly.

that's fine since that's relatively how it worked in mc live 2021 anyway (though they probably have it such that it only affect other shriekers within a 500 block radius or something)

**Comment by AViewFromTheTop** [comment]
@MerpisMe Just noticed something- the last vibration value is intentionally stored inside the Sensor's BlockEntity. That makes this a little easier!
I've gotta go for a bit, but I'm pasting the code used in the Sensor block to determine the last vibration frequency here so I can get back to it easier.

> BlockEntity blockEntity = world.getBlockEntity(pos);
>         if (blockEntity instanceof SculkSensorBlockEntity) {
>             SculkSensorBlockEntity sculkSensorBlockEntity = (SculkSensorBlockEntity)blockEntity;
>             return getPhase(state) == SculkSensorPhase.ACTIVE ? sculkSensorBlockEntity.getLastVibrationFrequency() : 0;
>         } else {
>             return 0;
>         }

**Comment by AViewFromTheTop** [comment]
Gotta go, everything should be done aside from bats. I'd appreciate it being tested as well

**Comment by MerpisMe** [comment]
> Gotta go, everything should be done aside from bats. I'd appreciate it being tested as well, I don't have time to test in-game so I'm hoping it works

just saw, i'll have to test after work-- amazing job, though, that was really quick!

**Comment by AViewFromTheTop** [comment]
> > Gotta go, everything should be done aside from bats. I'd appreciate it being tested as well, I don't have time to test in-game so I'm hoping it works
> 
> just saw, i'll have to test after work-- amazing job, though, that was really quick!

BTW, I'm not too sure if we should actually blacklist bats from triggering vibrations. There is a reason no mobs aside from the Warden spawn in the Deep Dark afterall.

**Comment by MerpisMe** [comment]
> > > Gotta go, everything should be done aside from bats. I'd appreciate it being tested as well, I don't have time to test in-game so I'm hoping it works
> > 
> > 
> > just saw, i'll have to test after work-- amazing job, though, that was really quick!
> 
> BTW, I'm not too sure if we should actually blacklist bats from triggering vibrations. There is a reason no mobs aside from the Warden spawn in the Deep Dark afterall.

yeah but my main concern was that bats would cause the warden to spawn without the player
